lib/dayron/request.ex

```diff
@@ -3,21 +3,109 @@ defmodule Dayron.Request do
   Defines a struct to store data necessary to send a request using an adapter.
   Also provides helper functions to map request data to adapter method calls.
   """
-  defstruct method: :get, url: "", body: %{}, headers: []
-  @type t :: %__MODULE__{method: atom, url: binary, body: map, headers: list}
+  defstruct method: :get, url: "", body: %{}, headers: [], options: []
+  @type t :: %__MODULE__{method: atom, url: binary, body: map, headers: list,
+                         options: list}
 
   @doc """
   Given a request struct and an adapter, calls the correct adapter function
   passing the correct parameters. Return a tuple with {request, response}.
   """
-  def send(request, adapter, opts \\ []) do
-    {_, response} = case request.method do
+  def send(request, adapter) do
+    start = current_time
+    opts = request.options
+    adapter_response = case request.method do
       :get -> adapter.get(request.url, request.headers, opts)
       :post -> adapter.post(request.url, request.body, request.headers, opts)
       :patch -> adapter.patch(request.url, request.body, request.headers, opts)
       :delete -> adapter.delete(request.url, request.headers, opts)
     end
 
-    {request, response}
+    {request, response_with_time(adapter_response, start)}
+  end
+
+  defp response_with_time({:ok, response}, start) do
+    %{response | elapsed_time: time_diff(start, current_time)}
+  end
+  defp response_with_time({_, response}, _), do: response
+  defp current_time, do: :os.timestamp()
+  defp time_diff(start, stop), do: :timer.now_diff(stop, start)
+end
+
+defimpl Inspect, for: Dayron.Request do
```

Comment:
since it's related to Request module, I think it's good to have everything in a single file..